src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js

```diff
@@ -0,0 +1,181 @@
+  const { succeeded, failed, running, submitted, ...others } = groupCallStatuses(calls)
+
+  const makeRow = (count, icon, text) => {
+    return !!count && div({ style: { display: 'flex', alignItems: 'center', marginTop: '0.25rem' } }, [
+      icon,
+      span([` ${count} ${text}`])
+    ])
+  }
+
+  return h(Fragment, [
+    makeRow(submitted, submittedIcon(), 'Submitted'),
+    makeRow(running, runningIcon(), 'Running'),
+    makeRow(succeeded, successIcon(), 'Succeeded'),
+    makeRow(failed, failedIcon(), 'Failed'),
+    _.map(([name, count]) => makeRow(count, unknownIcon(), name), _.toPairs(others))
+  ])
+}
+
+const WorkflowDashboard = _.flow(
+  Utils.forwardRefWithName('WorkflowDashboard'),
+  wrapWorkspace({
+    breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
+    title: 'Job History', activeTab: 'job history'
+  })
+)((props, ref) => {
+  const { namespace, name, submissionId, workflowId, workspace: { workspace: { bucketName } } } = props
+
+  /*
+   * State setup
+   */
+  const [workflow, setWorkflow] = useState({})
+  const [showLog, setShowLog] = useState(false)
+
+  const signal = Utils.useCancellation()
+  const stateRefreshTimer = useRef()
+
+  /*
+   * Data fetchers
+   */
+
+  Utils.useOnMount(() => {
+    const loadWorkflow = async () => {
+      const includeKey = [
+        'end', 'executionStatus', 'failures', 'start', 'status', 'submittedFiles:workflow', 'workflowLog', 'workflowName'
+      ]
+      const wf = await Ajax(signal).Workspaces.workspace(namespace, name).submission(submissionId).getWorkflow(workflowId, includeKey)
+      setWorkflow(wf)
+
+      if (_.includes(wf.status, ['Running', 'Submitted'])) {
+        stateRefreshTimer.current = setTimeout(loadWorkflow, 60000)
+      }
+    }
+
+    loadWorkflow()
+    return () => {
+      clearTimeout(stateRefreshTimer.current)
+    }
+  })
+
+  /*
+   * Page render
+   */
+  const { end, failures, start, status, workflowLog, workflowName, submittedFiles: { workflow: wdl } = {} } = workflow
+
+  const restructureFailures = failuresArray => {
+    const filtered = _.filter(({ message }) => !_.isEmpty(message) && !message.startsWith('Will not start job'), failuresArray)
+    const sizeDiff = failuresArray.length - filtered.length
+    const newMessage = sizeDiff > 0 ? [{ message: `${sizeDiff} jobs were queued in Cromwell but never sent to the cloud backend due to failures elsewhere in the workflow` }] : []
+    const simplifiedFailures = [...filtered, ...newMessage]
+
+    return _.map(({ message, causedBy }) => ({
+      message,
+      ...(!_.isEmpty(causedBy) ? { causedBy: restructureFailures(causedBy) } : {})
+    }), simplifiedFailures)
+  }
+
+  return div({ style: { padding: '1rem 2rem 2rem', flex: 1, display: 'flex', flexDirection: 'column' } }, [
+    workflowDetailsBreadcrumbSubtitle(namespace, name, submissionId, workflowId),
+    _.isEmpty(workflow) ? centeredSpinner() : div({ style: { display: 'flex', flexWrap: 'wrap' } }, [
```

Comment:
To allow us to use the standard pattern of undefined as not yet loaded:
```suggestion
    workflow === undefined ? centeredSpinner() : div({ style: { display: 'flex', flexWrap: 'wrap' } }, [
```

Reply:
I thought `isEmpty(workflow)` was a strictly more safe variant of `workflow === undefined` because it ALSO covered an empty object?

Reply:
Sure, the `=== undefined` is our standard pattern for "if this hasn't been loaded", though. The fact that `isEmpty` handles undefined is a convenience, it's not something we should consider semantic.